src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/CosmosDb/CosmosCommon.cs

```diff
@@ -0,0 +1,174 @@
+// <copyright file="CosmosCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.CosmosDb
+{
+    internal static class CosmosCommon
+    {
+        public const string MicrosoftAzureCosmosClientAssemblyName = "Microsoft.Azure.Cosmos.Client";
+        public const string MicrosoftAzureCosmosFeedIteratorTypeName = "Microsoft.Azure.Cosmos.FeedIterator`1<T>";
+        public const string MicrosoftAzureCosmosQueryDefinitionTypeName = "Microsoft.Azure.Cosmos.QueryDefinition";
+        public const string MicrosoftAzureCosmosQueryRequestOptionsTypeName = "Microsoft.Azure.Cosmos.QueryRequestOptions";
+        public const string Major3Minor6 = "3.6.0";
+        public const string Major3MinorX = "3";
+
+        public const string IntegrationName = nameof(IntegrationIds.CosmosDb);
+
+        private const string OperationName = "cosmosdb.query";
+        private const string ServiceName = "cosmosdb";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(CosmosCommon));
+
+        public static CallTargetState CreateContainerCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetContainerPropterties, instance, queryDefinition);
+        }
+
+        public static CallTargetState CreateDatabaseCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetDatabasePropterties, instance, queryDefinition);
+        }
+
+        public static CallTargetState CreateClientCallStateExt<TTarget, TQueryDefinition>(TTarget instance, TQueryDefinition queryDefinition)
+        {
+            return CreateCosmosDbCallState(GetClientPropterties, instance, queryDefinition);
+        }
+
+        private static CallTargetState CreateCosmosDbCallState<TTarget, TQueryDefinition>(Func<object, Tuple<string, string, string>> extractProperties, TTarget instance, TQueryDefinition queryDefinition)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
```

Comment:
We have some optimizations for this where you can use an `IntegrationInfo` instead of a string. IIRC the goal is to reduce heap allocations on every call to the integration.

Example usage:
https://github.com/DataDog/dd-trace-dotnet/blob/b68cdfef714b378cd61ced1fa2894e0e215d90a9/src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/GraphQL/GraphQLCommon.cs#L21-L22
https://github.com/DataDog/dd-trace-dotnet/blob/b68cdfef714b378cd61ced1fa2894e0e215d90a9/src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/GraphQL/GraphQLCommon.cs#L34

Reply:
Done.